NebulaWorld/SimulatedWorld.cs

```diff
@@ -144,5 +147,17 @@ public static void MineVegetable(VegeMined packet)
             }
         }
 
+        public static void OnGameLoadCompleted()
+        {
+            if (initialized == false)
+                return;
+
+            Log.Info("Game has finished loading");
+            //Assign our own coler
+            UpdatePlayerColor(GameMain.data.mainPlayer.transform,
+                new Float3(UnityEngine.Random.Range(0.0f, 1.0f), UnityEngine.Random.Range(0.0f, 1.0f), UnityEngine.Random.Range(0.0f, 1.0f)));
```

Comment:
Shouldn't you use the `LocalPlayer.Data.Color` instead of generating a new random color ?

Reply:
Nowhere else is the player assigned a random color (not as far as i can tell)

Reply:
It's done right now for remote client in the `NebulaHost.PlayerManager.PlayerConnected()`
And for the Host itself, it is in the `NebulaHost.MultiplayerHostSession.StartServer()`